metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -970,12 +979,13 @@ def _get_ge_dataset(
         if platform is not None and platform == "bigquery":
             # This is done as GE makes the name as DATASET.TABLE
             # but we want it to be PROJECT.DATASET.TABLE instead for multi-project setups
-            logger.debug(f"Setting table name to be {pretty_name}")
-            batch._table = sa.text(pretty_name)
             name_parts = pretty_name.split(".")
             if len(name_parts) != 3:
                 logger.error(
                     f"Unexpected {pretty_name} while profiling. Should have 3 parts but has {len(name_parts)} parts."
                 )
+            if len(str(batch._table).split(".")) == 2:
```

Comment:
Can you add comments on the PR for the special cases added? It is not clear why we have this special case